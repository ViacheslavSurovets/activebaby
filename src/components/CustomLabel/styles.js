import styled from 'styled-components';




export const Label = styled.label`
  position: absolute;
  left: 0;
  top: 0;
  transform: translateY(-100%);
  transition: 300ms ease all;
  margin-top: ${ props => props.marginTop }rem;
  margin-bottom: ${ props => props.marginBottom }rem;
  margin-right: ${ props => props.marginRight }rem;
  margin-left: ${ props => props.marginLeft }rem;
  font-size: ${ props => props.fontSize }rem;
`;
