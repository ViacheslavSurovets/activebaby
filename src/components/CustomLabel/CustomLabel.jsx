import React from 'react';
import { Label } from './styles';
import PropTypes from 'prop-types';

const CustomLabel = ( { children , ...rest } ) => (
  <Label { ...rest }>
    { children }
  </Label>
);

export default CustomLabel;

CustomLabel.propTypes = {
  children: PropTypes.string
};
