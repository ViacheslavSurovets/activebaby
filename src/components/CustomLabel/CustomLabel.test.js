import React from 'react';
import { shallow } from 'enzyme';
import CustomLabel from './CustomLabel';


describe ( 'CustomLabel', () => {
  it ( 'CustomLabel Snapshot', () => {
    expect ( shallow ( <CustomLabel /> ) ).toMatchSnapshot ();
  } );
} );
