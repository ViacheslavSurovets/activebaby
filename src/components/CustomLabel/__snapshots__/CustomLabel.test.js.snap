// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomLabel CustomLabel Snapshot 1`] = `ShallowWrapper {}`;
